### PR TITLE
`[Exiled::API]` New way to register parent commands

### PR DESCRIPTION
**Pull request by VALERA771**
## Description
**Describe the changes** 
Adding a new way to register commands for parent commands. Iirc this is system which is similar to nw

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?** (if this is a feature change)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:
No breaking changes cuz both systems are supported (new and current)
~(but for me this is better)~

<br />

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [ ] I have checked no IL patching errors in the console

### Other
- [x] Still requires more testing (i would like if someone else will also test it)

### CODE REVIEW

**Comment by VALERA771** [comment]
Tested on exiled commands but as I said it will be great if someone else would also test
TL;DR yeah it's tested

**Review by VladTheCow — COMMENTED** [review]
I honestly don't like the idea of it because it breaks the logic of base game command handler API

**Comment by VALERA771** [comment]
Idk how it brokes it. As i said I've heard something (this information might be a lie) that game is using same logic

**Comment by louis1706** [comment]
> I honestly don't like the idea of it because it breaks the logic of base game command handler API

it's how nw register their command just than they do it in Compilation instead

**Review by Jesus-QC — APPROVED** [review]

**Review by VladTheCow — APPROVED** [review]
